onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <ctime>
+#include "core/common/logging/logging.h"
+#include "core/framework/op_kernel.h"
+#include "NvInfer.h"
+#include "NvOnnxParser.h"
+
+namespace onnxruntime {
+
+static const int kMaxBatchSize = 16;
+static const int kMaxWorkSpaceSize = 16 << 20;
+
+#define CHECK_CUDA(call)                         \
+  do {                                           \
+    cudaError_t status = call;                   \
+    if(status != cudaSuccess) {                  \
+      return -1;                                 \
+    }                                            \
+  } while(0)
+
+struct InferDeleter {
+  template <typename T>
+  void operator()(T* obj) const {
+    if (obj) {
+      obj->destroy();
+    }
+  }
+};
+
+template <typename T>
+using unique_pointer = std::unique_ptr<T, InferDeleter>;
```

Comment:
>unique_pointer [](start = 6, length = 14)

This should also be better scoped. I think having onnxruntime::unique_pointer being specific to InferDeleter would be fairly unexpected.

Reply:
The unique_pointer is also used to define private member. that's why I keep it in the header.